### DISCUSSION TITLE
Issue(s) while using images with Survey Likert Plugin

### DISCUSSION THREAD

**Discussion by batshell** [jspsych/jsPsych#1609] [Q&A] [answered]
So, I have to create an Image rating task and I need to hide the image after 5 seconds. Here are the files that I have edited:

added stim duration in the jspsych-survey-likert.js file:
``` 
stimulus_duration: {
        type: jsPsych.plugins.parameterType.INT,
        pretty_name: 'Stimulus duration',
        default: null,
        description: 'How long to show the stimulus.'
      },
```

In the main experiment file, I have added a new div tag for image with id: imgratehide and to hide it, I have added this code:
```
if (trial.stimulus_duration !== null) {
      jsPsych.pluginAPI.setTimeout(function() {
        document.getElementById('imgratehide').style.visibility = 'hidden';
      }, trial.stimulus_duration);
    }
```

Problem(s):
1. Images take about 1-2 seconds to load, even when the images are small (max 30KB) and I also tried preload.
2. Sometimes the images disappear before the set time limit, to control this, I had to add a blank trial (200ms) between each of the rating task.

**Comment by jodeleeuw** [comment, marked as answer]
Hi @batshell,

> Images take about 1-2 seconds to load, even when the images are small (max 30KB) and I also tried preload.

That's odd. Can you share more about the preloading approach you used? Are you using the `preload` plugin with `6.3.0`?

> Sometimes the images disappear before the set time limit, to control this, I had to add a blank trial (200ms) between each of the rating task.

Does this happen only after finishing a trial quickly? The `setTimeout` handler might still be running, and affect your next trial. You may need to call `jsPsych.pluginAPI.cancelAllTimeouts()` before ending the trial.

**Reply by batshell** [reply]
For preload, I used the preload plugin and added this code after the script tag in main expt file:
```
var preload = {
    type: 'preload',
    auto_preload: true
};
```

Can you tell more about how to use timeout function? I tried adding this function with on_finish: in survey likert but it's not working.

**Reply by jodeleeuw** [reply]
The `auto_preload: true` option won't work in this case because the images you are using aren't part of image-specific parameters (like `stimulus` in the `image-keyboard-response` plugin). When you use images in HTML markup jsPsych can't automatically detect those paths, so you'll need to load the images manually. Try using:

```js
var preload = {
	type: 'preload',
	images: [...]
}
```

**Reply by batshell** [reply]
Thank you, but I had to preload all the images in a hidden "div" before start of the experiment.

**Reply by jodeleeuw** [reply]
That shouldn't be necessary to do. Did providing a list of the images to the preload plugin not work? If there's a bug I'd love to track it down.

**Reply by batshell** [reply]
> That shouldn't be necessary to do. Did providing a list of the images to the preload plugin not work? If there's a bug I'd love to track it down.

Yea, it did not work, I did not see any error in the console as well.

**Reply by batshell** [reply]
@jodeleeuw I tried the preloading again by following the tutorial line by line, and it works great!
(https://www.jspsych.org/overview/media-preloading/)
Maybe I was doing something wrong, you can mark this thread as solved.
